components/core/src/clp_s/Utils.hpp

```diff
@@ -264,10 +267,87 @@ class Span {
 
     T& operator[](size_t i) { return m_begin[i]; }
 
+    Span<T> sub_span(size_t start, size_t size) { return {m_begin + start, size}; }
+
 private:
     T* m_begin;
     size_t m_size{};
 };
 }  // namespace clp_s
 
+template <typename T>
+class UnalignedSpan {
+public:
+    UnalignedSpan() = default;
+    UnalignedSpan(char* begin, size_t size) : m_begin(begin), m_size(size){};
+
+    size_t size() { return m_size; }
+
+    size_t size_in_bytes() { return m_size * sizeof(T); }
+
+    T operator[](size_t i) {
+        T tmp;
+        memcpy(&tmp, m_begin + i * sizeof(T), sizeof(T));
+        return tmp;
+    }
+
+    UnalignedSpan<T> sub_span(size_t start, size_t size) {
+        return {m_begin + start * sizeof(T), size};
+    }
+
+private:
+    char* m_begin{nullptr};
+    size_t m_size{0};
+};
+
+/**
+ * ManagedBufferViewReader is a utility class designed to let several readers safely share views
+ * into a shared buffer. Readers can consume from the buffer either by consuming spans and scalar
+ * values from the buffer. Any attempt to consume past the end of the buffer will throw.
+ * @throws OperationFailed
+ */
+class ManagedBufferViewReader {
+public:
+    class OperationFailed : public clp_s::TraceableException {
+    public:
+        // Constructors
+        OperationFailed(clp_s::ErrorCode error_code, char const* const filename, int line_number)
+                : TraceableException(error_code, filename, line_number) {}
+    };
+
+    explicit ManagedBufferViewReader(char* buffer, size_t size)
+            : m_buffer(buffer),
+              m_remaining_size(size) {}
+
+    template <typename T>
+    T consume_value() {
+        if (m_remaining_size < sizeof(T)) {
+            throw OperationFailed(clp_s::ErrorCodeOutOfBounds, __FILENAME__, __LINE__);
+        }
+        T tmp;
+        memcpy(&tmp, m_buffer, sizeof(T));
+        m_buffer += sizeof(T);
+        m_remaining_size -= sizeof(T);
+        return tmp;
+    }
+
+    template <typename T>
+    UnalignedSpan<T> consume_unaligned_span(size_t length) {
```

Comment:
Do you want to add a description for this method? I'm fine without it since it's relatively straightforward.

Reply:
Yeah, after renaming to read_xyz I think it's self documenting enough